clmul/src/backend.rs

```diff
@@ -56,103 +56,94 @@ cfg_if! {
     }
 }
 
-cfg_if! {
-    if #[cfg(any(all(target_arch = "aarch64", feature = "armv8"), any(target_arch = "x86_64", target_arch = "x86")))]{
-        #[derive(Clone, Copy, Debug)]
-        /// Carryless multiplication
-        pub struct Clmul {
-            intrinsics: Option<intrinsics::Clmul>,
-            soft: Option<soft::Clmul>,
+use lazy_static::lazy_static;
+lazy_static! {
+    static ref IS_INTR: bool = {
+        cfg_if! {
+            if #[cfg(feature = "force-soft")] {
+                false
+            } else if #[cfg(any(all(target_arch = "aarch64", feature = "armv8"), any(target_arch = "x86_64", target_arch = "x86")))]{
+                mul_intrinsics::get()
+            } else {
+                // "force-soft" feature was not enabled but neither was
+                //  supported arch found. Falling back to soft backend.
+                false
+            }
```

Comment:
I also tried to follow the polyval implementation but was slightly concerned that if we use `token` then it becomes difficult to deal with the feature `force-soft`. Shall I just use `token`? @sinui0

Reply:
Alright I updated in c0c846d you can check out